Setup.md

```diff
@@ -181,3 +181,45 @@ See [c extensions](./mqttclients/c/README.md) for more details.
 ### Python
 
 Python samples have been tested with python 3.10.4, to install follow the instructions from https://www.python.org/downloads/ 
+
+### TypeScript
+
+TypeScript samples have been tested with NodeJS version 18.16.0 and NPM version 9.5.1. Version 18 or higher of NodeJS and version 8 or higher of NPM is required. See https://nodejs.org. The samples are written using the [MQTT.js library](https://www.npmjs.com/package/mqtt).
+
+The TypeScript samples are built using [TypeScript ESLint](https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/) for Visual Studio Code and [TypeScript project references](https://www.typescriptlang.org/docs/handbook/project-references.html). This allows abstracted client and utility classes to be separate dependent projects of the main example scenario projects.
+
+If running a sample that uses protobuf - currently these are the Command Samples. Note that you'll need the protobuf compiler if you are generating code for new proto files. The TypeScript samples use [protoc](https://grpc.io/docs/protoc-installation/) for the compiler and [ts-protoc-gen](https://www.npmjs.com/package/ts-protoc-gen?activeTab=readme) for the TypeScript types.
```

Comment:
can we install this from package.json?

Reply:
I don't think so, maybe running a script during install? But, I'm not sure if that is anti-pattern. I took a hint from the C samples which require some pre-installed binaries.

Reply:
at least add the command, to save the user one visit to the typescript site :)

Reply:
Done.